gridstatus/spp.py

```diff
@@ -193,6 +201,189 @@ def get_load_forecast(self, date, forecast_type="MID_TERM", verbose=False):
 
         return current_day_forecast
 
+    def get_solar_and_wind_load_forecast(self, date, end=None, verbose=False):
+        """Solar and wind forecast for upcoming days. In 5-minute intervals if end
+        is less than 4 hours from now. Otherwise in hourly intervals.
+
+        Returns:
+            pd.DataFrame: forecast for current day
+        """
+        now = pd.Timestamp.now(tz=self.default_timezone)
+
+        use_shortterm = False
+
+        # Only get the shortterm forecast if end is provided and end is less
+        # than 4 hours from now
+        if end:
+            end = utils._handle_date(end, self.default_timezone)
+            use_shortterm = end < now + pd.Timedelta(hours=4)
+            # Files don't exist in the future, so we need to limit the end.
+            end = min(end, now)
+
+        if date == "latest":
+            date = now
+
+        date = utils._handle_date(date, self.default_timezone)
+
+        if use_shortterm:
+            df = self._get_5_min_solar_and_wind_forecast_and_actuals(
+                date.floor("5T"),
+                end,
+                verbose=verbose,
+            )
+        else:
+            if date > now + pd.Timedelta(days=7):
+                raise ValueError("Forecast only available for next 7 days")
+
+            df = self._get_hourly_solar_and_wind_forecast(
+                date.floor("H"),
+                end,
+                verbose=verbose,
+            )
+
+        df.columns = [col.strip() for col in df.columns]
+
+        # Don't include the actuals to avoid confusion
+        cols_to_drop = [col for col in df if "Actual" in col]
```

Comment:
i think we actually should keep the actuals in the same file. just to make it clear where they are coming from

i took a peak at the data and the actuals seem to different the number reported in this report: https://portal.spp.org/pages/generation-mix-historical